src/pipdeptree/_non_host.py

```diff
@@ -41,7 +42,9 @@ def handle_non_host_target(args: Options) -> int | None:
             copytree(our_root, dest / "pipdeptree")
             # invoke from an empty folder to avoid cwd altering sys.path
             env = os.environ.copy()
-            env["PYTHONPATH"] = project
+            # add dist path of packaging to PYTHONPATH, which may not in given env.
+            dist = Distribution.from_name("packaging")
+            env["PYTHONPATH"] = project + ":" + str(dist._path)
```

Comment:
Sadly this won't work since `dist._path` points to the .dist-info directory. We could just do `Distribution.locate_file("") / packaging` but I think the following way is better (I just snatched this idea from reading the code above it):

```suggestion
            # We need to also make the Python executable aware of packaging since we use it.
            packaging_src = getsourcefile(sys.modules["packaging"])
            assert packaging_src is not None
            packaging_root = Path(packaging_src).parent
            copytree(packaging_root, dest / "packaging")
```

It recursively copies the contents of `site-packages/packaging` to the temporary directory we use to also host the pipdeptree source code. It's not robust, but it doesn't make sense for `packaging` not to exist if we are executing `pipdeptree`.

Reply:
Yes, `Distribution.locate_file("") / packaging`is the most direct way. I made some attempts, but you are right, it doesn't work...
But I didn't get that why the latter is better. It does more works with copying source dir which we can just use.

Reply:
Oh, I seem to know.
Adding `Distribution.locate_file("") / packaging` to `PYTHONPATH` doesn't work either.
What should be added is `Distribution.locate_file("")`, not his sub-directory.